[Feature] 링크 추가 UI 구성

joseph704 opened the pull request:
## 📌 배경

close #26 

## 내용
- 링크 추가 UI 구성
- 여백 클릭 시, 키보드 내리기 위한 함수 추가

## 테스트 방법 (optional)
```swift
     LinkAddView(fromWhere: .naviBar)
     LinkAddView(fromWhere: .snackBar)
```

## 스크린샷 (optional)
|링크없을때|링크 있을때|아티클 제목 수정중|메모 수정중|공개설정 툴팁|
|:-:|:-:|:-:|:-:|:-:|
| ![IMG_6795](https://user-images.githubusercontent.com/35060252/206969850-cc6d0d3d-5b1e-4bc3-8518-764f410ae200.PNG) |  ![IMG_6796](https://user-images.githubusercontent.com/35060252/206969903-32a97d7a-6475-4cb7-bf7a-62d63963ffd5.PNG) | ![IMG_6797](https://user-images.githubusercontent.com/35060252/206969927-a01845bc-2c8e-419c-b611-4c8b3adfa473.PNG) | ![IMG_6798](https://user-images.githubusercontent.com/35060252/206969951-b7e533f9-a03a-44f0-9430-939ba5ddd966.PNG) | ![IMG_6799](https://user-images.githubusercontent.com/35060252/206969978-57994932-4b26-4fc4-a2e8-3ca6753111ca.PNG) |

joseph704 committed:
feat: 여백 클릭 시, 키보드 내리기 위한 함수 추가
#26

joseph704 committed:
feat: 링크 추가 UI 구성
#26

joseph704 committed:
feat: 링크 추가 화면을 띄운 버튼이 snackBar인 경우, 복사한 텍스트를 링크 textField에 자동으로 세팅되도록 수정
#26

joseph704 committed:
feat: 홈 화면에서 링크 추가 화면을 띄우도록 구현
#26
- 하위뷰인 HomeView에서 NavigationView를 지우고, 상위뷰인 MainTabView에 NavigationView를 삽입함 (MainTabView에서 NavigationLink로 링크 추가 View를 띄우기 위해)
- snackBar or naviBar 통해 링크 추가 화면 이동하도록 함

kipsong133 reviewed (approved):
이미지 추가하신 부분이랑, 키보드 관련 제스처 유용하게 사용할게요~! 
내용 확인했습니다!

Marcorable commented on Targets/UserInterface/Sources/LinkAdd/View/LinkAddView.swift:
```diff
@@ -0,0 +1,265 @@
+import Combine
+
+enum FromWhichButton {
+    case naviBar
```
진짜 별거 아니긴 한데.. 질문입니다!
탭바 중앙에 버튼으로 들어온 경우인데 네비로 지으신 이유가 혹시 있으신가요?
NavigationLink로 들어와서 일까요?

joseph704 replied:
@keeplo 
피그마에 제시된대로 작명하였습니다. 추후에 디자이너 분들과의 공통 용어로 쓰기 좋도록 그대로 이행해봤습니다 ㅎㅎ
<img width="429" alt="image" src="https://user-images.githubusercontent.com/35060252/207075813-2a29f2e8-e739-4d27-a847-2c44eaf8e4af.png">

Marcorable replied:
답변감사합니다!

Marcorable commented on Targets/UserInterface/Sources/LinkAdd/View/LinkAddView.swift:
```diff
@@ -0,0 +1,265 @@
+    case memo = "메모"
+}
+
+struct LinkAddView: View {
```
타입 네이밍에 동사형은 조금 피하면 좋을것 같아요

프리뷰처럼 `Adding`을 추천드려봅니다...

joseph704 replied:
앗.. 듣고보니 그러네요.. 수정하도록 하겠습니다. 감사합니다! (영어 스타 강사의 짬바란..)

Marcorable replied:
그쵸 서당개 1년 입니다 ㅎ

joseph704 replied:
@keeplo 
해당 부분 수정했습니다. 확인부탁드립니다!

Marcorable commented on Targets/UserInterface/Sources/LinkAdd/View/LinkAddView.swift:
```diff
@@ -0,0 +1,265 @@
+                            .padding(.vertical, 14)
+                        Button {
+                            isPublicSettingsHelpClicked.toggle()
+                        } label: {
+                            Image("CloseButton")
+                                .frame(width: 16.0, height: 16.0)
+                                .padding(.trailing, 18.34)
+                                .padding(.top, 14)
+                        }
+                    }
+                    .frame(height: 70)
+                    .background(Color.grey2Line)
+                    .cornerRadius(4.0)
+                    .overlay(
+                        RoundedRectangle(cornerRadius: 4.0)
+                            .stroke(Color.ticlemoaBlack, lineWidth: 1.0)
+                        )
+                    Spacer()
+                }
+                .padding(.trailing, 32)
+            }
+            Spacer()
+            Button {
+                print("추가하기 버튼!")
+            } label: {
+                Text("추가하기")
+                    .tint(viewModel.link.isEmpty ? .grey3 : .ticlemoaWhite)
+            }
+            .frame(maxWidth: .infinity)
+            .frame(height: 56)
+            .background(viewModel.link.isEmpty ?  Color.grey2Line : Color.ticlemoaBlack)
+            .cornerRadius(6.0)
+            .padding(.bottom, 16)
+        }
+        .hideKeyboard()
+        .padding(.horizontal, 20)
+        .navigationTitle("링크 추가")
+        .navigationBarTitleDisplayMode(.inline)
+    }
+}
```
워낙 한 화면에 구성되는 게 많아서 어쩔 수 없는 부분이긴한데..

연산 프로퍼티로 단락을 나눠서
`body` 에 View 구성을 파악하기 쉽게 나누면 어떨까요?
코드 가독성이 쪼끔 더 좋아지지 않을까 생각합니다!

joseph704 replied:
@keeplo 
감사합니다. 한번 나눠 보겠습니다! 

예를들어 아래 표시한 빨간 원 마다 이해하기 쉬운 View(`ex.첫번째 빨간 원 == InsertingLinkView`)로 묶어서 body에 두면 나을 것 같다는 말씀이시죠?

<img width="346" alt="image" src="https://user-images.githubusercontent.com/35060252/207077404-c0d0de25-83db-4e84-b757-03de05cee1b8.png">

Marcorable replied:
넵 말씀해주신 방향이 맞습니다!

joseph704 replied:
@keeplo 
해당 부분 수정했습니다. 확인부탁드립니다.

Marcorable reviewed (changes requested):
화면에 뭐가 진짜 많아서 복잡 했을텐데 고생하셨습니다.
키보드 내리기 Extension 도 정말 감사합니당

sustainable-git commented on Targets/UserInterface/Sources/Home/HomeView.swift:
```diff
@@ -14,30 +14,8 @@ struct HomeView: View {
     }
     
     var body: some View {
-        NavigationView {
```
NavigationView가 없어도 되나요?
HomeView에서 ProfileView로 넘어갈 때 필요할것 같습니다.

joseph704 replied:
@sustainable-git 
HomeView 뿐만 아니라 다른곳에서 사용해야할 것 같아서, HomeView의 상위뷰인 MainTabView에 NavigationView추가해 두었습니다!

joseph704 commented on Targets/UserInterface/Sources/MainTab/MainTabView.swift:
```diff
@@ -9,31 +9,89 @@
-                .tabItem {
-                    Tab.home.imageItem
-                    Tab.home.textItem
+        NavigationView {
```
@sustainable-git 
NavigationView 여기에 추가해 두었습니다!

joseph704 committed:
feat: 링크 추가 네이밍 변경
#26
 - LinkAddView -> AddingLinkView, LinkAddViewModel -> AddingLinkViewModel로 변경

joseph704 committed:
feat: 링크추가 화면 구성 요소별로 연산 프로퍼티로 나눠서 구성

Marcorable reviewed (approved):
늦어서 죄송해용 고생하셨습니다!
